pytorch_lightning/plugins/precision/hpu.py

```diff
@@ -0,0 +1,39 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any, Optional, Sequence
+
+from pytorch_lightning.plugins.precision.precision_plugin import PrecisionPlugin
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from pytorch_lightning.utilities.imports import _HPU_AVAILABLE
+
+if _HPU_AVAILABLE:
+    from habana_frameworks.torch.hpex import hmp
+
+
+class HPUPrecisionPlugin(PrecisionPlugin):
+    """Plugin that enables bfloats/floats on HPUs."""
+
+    def __init__(self, precision: int, hmp_params: Optional[Sequence[Any]] = None) -> None:
```

Comment:
```suggestion
    def __init__(
        self,
        precision: int,
        level: str = "02",
        bf16_ops: Optional[str] = None,
        bf32_ops: Optional[str] = None,
        verbose: bool = False
    ) -> None:
```

Reply:
I would prefer to flatten the arguments. This would make it more discoverable and readable.

Reply:
After offline discussion, we came to the conclusion that for now as habana sdk is in the experimental state and new features for fp16, etc will be added soon, user could pass `hmp_params` dictionary rather than us bloating up the plugin interface and keep updating it

Reply:
Sounds good with me !